servicetalk-http-netty/src/test/java/io/servicetalk/http/api/ServiceTalkToNettyContentCodingCompatibilityTest.java

```diff
@@ -0,0 +1,167 @@
+    public void finish() throws Exception {
+        serverAcceptorChannel.close().syncUninterruptibly();
+        serverEventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS).syncUninterruptibly();
+        client.close();
+    }
+
+    private Channel newNettyServer() {
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(serverEventLoopGroup);
+        sb.channel(serverChannel(serverEventLoopGroup, InetSocketAddress.class));
+
+        sb.childHandler(new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(final Channel ch) {
+                ChannelPipeline p = ch.pipeline();
+                p.addLast(new HttpServerCodec());
+                if (!scenario.serverSupported.isEmpty()) {
+                    p.addLast(new HttpContentDecompressor());
+                    p.addLast(new HttpContentCompressor());
+                }
+                p.addLast(EchoServerHandler.INSTANCE);
+            }
+        });
+        return sb.bind(localAddress(0)).syncUninterruptibly().channel();
+    }
+
+    @Override
+    public void testCompatibility() throws Exception {
+        assumeFalse("Only testing H1 scenarios yet.", scenario.isH2);
+        assumeTrue("Only testing successful configurations; Netty doesn't have knowledge " +
+                "about unsupported compression types.", scenario.valid);
+
+        super.testCompatibility();
+    }
+
+    @Override
+    protected HttpClient client() {
+        return client;
+    }
+
+    @ChannelHandler.Sharable
+    static class EchoServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+        static final EchoServerHandler INSTANCE = new EchoServerHandler();
+
+        private static final byte[] CONTENT = payload((byte) 'b');
+
+        @Override
+        public void channelReadComplete(ChannelHandlerContext ctx) {
+            ctx.flush();
+        }
+
+        @Override
+        public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
+            if (msg instanceof io.netty.handler.codec.http.HttpRequest) {
+                io.netty.handler.codec.http.HttpRequest req = (io.netty.handler.codec.http.HttpRequest) msg;
+
+                boolean keepAlive = HttpUtil.isKeepAlive(req);
+                FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(),
+                        OK, wrappedBuffer(CONTENT));
+
+                response.headers()
+                        .set(CONTENT_TYPE, TEXT_PLAIN)
+                        .setInt(CONTENT_LENGTH, response.content().readableBytes());
+
+                if (keepAlive) {
+                    if (!req.protocolVersion().isKeepAliveDefault()) {
+                        response.headers().set(CONNECTION, KEEP_ALIVE);
+                    }
+                } else {
+                    response.headers().set(CONNECTION, CLOSE);
+                }
+
+                ChannelFuture f = ctx.write(response);
+
+                if (!keepAlive) {
```

Comment:
For the purpose of the test, you don't need keep-alive logic. Just leave the connection open (`@After` will shutdown everything) or always close it, doesn't matter.